lib/pure/memfiles.nim

```diff
@@ -225,13 +226,16 @@ proc open*(filename: string, mode: FileMode = fmRead,
       result.size = mappedSize
     else:
       var stat: Stat
-      if fstat(result.handle, stat) != -1:
+      if fstat(result.handle, stat) != -1 and int(stat.st_size) > 0:
         # XXX: Hmm, this could be unsafe
         # Why is mmap taking int anyway?
         result.size = int(stat.st_size)
       else:
         fail(osLastError(), "error getting file size")
 
+    if result.size == 0:
```

Comment:
Why is this required? ``osLastError()`` is to be used when an OS operation failed (sets errno etc).

Reply:
You're correct. Changed to ValueError